rust/src/io/object_store.rs

```diff
@@ -223,6 +224,21 @@ impl ObjectStore {
     pub async fn size(&self, path: &Path) -> Result<usize> {
         Ok(self.inner.head(path).await?.size)
     }
+
+    /// Deletes a Path from the underlying storage
+    pub async fn delete(&self, path: &Path) -> Result<()> {
+        self.inner.delete(path).await?;
+        Ok(())
+    }
+
+    /// Deletes a directory from the underlying storage, only supported in local file systems
+    pub async fn remove_dir(&self, path: &Path) -> Result<()> {
+        if self.is_local() {
+            // FIXME cross-platform
+            std::fs::remove_dir(format!("/{}", path.to_string()))?
+        }
+        Ok(())
```

Comment:
Here you will finish the deletion on s3/gcs as well, right?

Reply:
Not necessary everything is tracked in manifest ?  Just deleting from what is referred from manifest does not seem safe to me.